patches/api/0480-Fix-HelpCommand-searching.patch

```diff
@@ -0,0 +1,41 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tamion <70228790+notTamion@users.noreply.github.com>
+Date: Sat, 13 Jul 2024 18:00:37 +0200
+Subject: [PATCH] Fix HelpCommand searching
+
+
+diff --git a/src/main/java/org/bukkit/command/defaults/HelpCommand.java b/src/main/java/org/bukkit/command/defaults/HelpCommand.java
+index a1699aa93c62f70238822c446326ed46ea71db14..c4f33220225141ca187b0d22c6d1e0ddbd7a8092 100644
+--- a/src/main/java/org/bukkit/command/defaults/HelpCommand.java
++++ b/src/main/java/org/bukkit/command/defaults/HelpCommand.java
+@@ -146,7 +146,7 @@ public class HelpCommand extends BukkitCommand {
+         for (HelpTopic topic : Bukkit.getServer().getHelpMap().getHelpTopics()) {
+             String trimmedTopic = topic.getName().startsWith("/") ? topic.getName().substring(1) : topic.getName();
+ 
+-            if (trimmedTopic.length() < searchString.length()) {
++            if (searchString.isEmpty() || trimmedTopic.length() < searchString.length()) { // Paper - Fix HelpCommand searching
+                 continue;
+             }
+ 
+diff --git a/src/main/java/org/bukkit/help/IndexHelpTopic.java b/src/main/java/org/bukkit/help/IndexHelpTopic.java
+index 88a812d66b7eea7389103b9c2776a8adcfc141c7..ea5d63e85abfa224c7fee025ad904cb56e74e4d7 100644
+--- a/src/main/java/org/bukkit/help/IndexHelpTopic.java
++++ b/src/main/java/org/bukkit/help/IndexHelpTopic.java
+@@ -49,10 +49,13 @@ public class IndexHelpTopic extends HelpTopic {
+         if (sender instanceof ConsoleCommandSender) {
+             return true;
+         }
+-        if (permission == null) {
+-            return true;
+-        }
+-        return sender.hasPermission(permission);
```

Comment:
I am unsure how clever it is to not check the permission here at all.
While nothing currently sets a permission, it is technically still a valid thing to do unless I am missing something?

The HelpMap is public API, so are the help topics.
Plugins making use of such a permission flag should be respected.

Reply:
I find it weird Bukkit has this IndexHelpTopic in the first place. What is the advantage against just using a List?

Reply:
Yea sadly we cannot crystal ball into the minds of bukkit developers xD